### PR TITLE
support sql forms and cases in related doc lookups

### PR DESCRIPTION
**Pull request by czue**
@dimagi/scale-team cc @nickpell 

:tropical_fish: the functional change happens in the last commit

### CODE REVIEW

File: corehq/apps/userreports/tests/test_expressions.py [hunk 1]
```diff
@@ -692,6 +695,55 @@ def test_caching(self):
...
+        case_id = uuid.uuid4().hex
+        create_and_save_a_case(domain=self.domain, case_id=case_id, case_name='related doc test case')
+        expression = self._get_expression('CommCareCase')
+        doc = self._get_doc(case_id)
+        self.assertEqual(case_id, expression(doc, EvaluationContext(doc, 0)))
+
+    @run_with_all_backends
+    def test_other_lookups(self):
+        user_id = uuid.uuid4().hex
+        user = CommCareUser.get_db().save_doc({'_id': user_id, 'domain': self.domain})
```
Comment (dimagimon) [inline corehq/apps/userreports/tests/test_expressions.py]: F841 local variable 'user' is assigned to but never used

**Comment by czue** [comment]
(wait for tests - incorrectly flagged as pasing now)

**Comment by snopoke** [comment]
:+1: on :white_check_mark:

**Comment by czue** [comment]
:white_check_mark:

**Pull request merged**